artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/RefsOperation.java

```diff
@@ -152,14 +153,14 @@ protected void rollbackRedelivery(Transaction tx, MessageReference ref, long tim
    @Override
    public void afterCommit(final Transaction tx) {
       for (MessageReference ref : refsToAck) {
-         synchronized (ref.getQueue()) {
-            queue.postAcknowledge(ref);
-         }
+         queue.postAcknowledge(ref);
       }
 
       if (pagedMessagesToPostACK != null) {
          for (MessageReference refmsg : pagedMessagesToPostACK) {
-            decrementRefCount(refmsg);
```

Comment:
I agree that right now it doesn't seem to be of much use except for large messages.  However if this gets taken out I would be a little concerned about introducing a future bug if other message types end up relying on this in the future.

Reply:
@cshannon thanks.